rpcq/_client.py

```diff
@@ -56,18 +57,21 @@ def __init__(self, endpoint: str, timeout: float = None):
         # Cache of replies so that different tasks can share results with each other
         self._replies: Dict[str, Union[RPCReply, RPCError]] = {}
 
-    async def call_async(self, method_name: str, *args, **kwargs):
+    async def call_async(self, method_name: str, timeout: float = None, *args, **kwargs):
         """
         Send JSON RPC request to a backend socket and receive reply (asynchronously)
 
         :param method_name: Method name
+        :param float timeout: Timeout in seconds for Server response, set to None to disable the timeout
         :param args: Args that will be passed to the remote function
         :param kwargs: Keyword args that will be passed to the remote function
         """
-        if self.timeout:
+        if not timeout:
+            timeout = self.timeout
+        if timeout:
```

Comment:
It looks like this `if` statement will always evaluate to True, so we shouldn't need the `if` `else` block.

Reply:
Not if someone patches out the `timeout`